airflow/api/common/trigger_dag.py

```diff
@@ -104,6 +107,7 @@ def _trigger_dag(
 
 def trigger_dag(
     dag_id: str,
+    triggered_by: DagRunTriggeredByType,
```

Comment:
This should probably be a keyword-only argument instead. Sticking a positional parameter in the middle is too easy to get wrong. Same applies for other functions that add this argument.

Reply:
OK, it makes sense. Do you think that it is wise to use `*` as a parameter before the `triggered_by` parameter to force to use named arguments (as below)? 
```python
def trigger_dag(
    dag_id: str,
    *,  # <- new line
    triggered_by: DagRunTriggeredByType,
```

I think the other option is setting to `None` as the default value and checking the value is `None` in the function body.  This option will definitely look ugly.

WDYT? @uranusjr

Reply:
Adding the star is the only way. Simply adding a default does not even make the argument *keyword-only*.